core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala

```diff
@@ -306,148 +299,83 @@ class ConsumerBounceTest extends BaseRequestTest with Logging {
     this.consumerConfig.setProperty(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, "60000")
     this.consumerConfig.setProperty(ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG, "1000")
     this.consumerConfig.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false")
-    val producer = createProducer()
-    createTopic(topic, numPartitions = partitionCount, replicationFactor = numBrokers)
-    val stableConsumers = createConsumersWithGroupId("group2", consumerCount, executor, topic = topic)
+    val partitions = createTopicPartitions(topic, numPartitions = partitionCount, replicationFactor = numBrokers)
 
-    // assert group is stable and working
-    sendRecords(producer, recordsProduced, topic, numPartitions = Some(partitionCount))
-    stableConsumers.foreach { cons => {
-      receiveAndCommit(cons, recordsProduced / consumerCount, 10000)
-    }}
+    addConsumersToGroupAndWaitForGroupAssignment(consumerCount, mutable.Buffer[KafkaConsumer[Array[Byte], Array[Byte]]](),
+      consumerPollers, List[String](topic), partitions, group)
 
     // roll all brokers with a lesser max group size to make sure coordinator has the new config
     val newConfigs = generateKafkaConfigs(maxGroupSize.toString)
-    val kickedConsumerOut = new AtomicBoolean(false)
+    var kickedConsumerOut = false
     var kickedOutConsumerIdx: Option[Int] = None
-    val lock = new ReentrantLock
     // restart brokers until the group moves to a Coordinator with the new config
     breakable { for (broker <- servers.indices) {
       killBroker(broker)
-      sendRecords(producer, recordsProduced, topic, numPartitions = Some(partitionCount))
-
-      var successfulConsumes = 0
-
-      // compute consumptions in a non-blocking way in order to account for the rebalance once the group.size takes effect
-      val consumeFutures = new ArrayBuffer[SFuture[Any]]
-      implicit val executorContext: ExecutionContextExecutor = ExecutionContext.fromExecutor(executor)
-      stableConsumers.indices.foreach(idx => {
-        val currentConsumer = stableConsumers(idx)
-        val consumeFuture = SFuture {
-          try {
-            receiveAndCommit(currentConsumer, recordsProduced / consumerCount, 10000)
-            CoreUtils.inLock(lock) { successfulConsumes += 1 }
-          } catch {
-            case e: Throwable =>
-              if (!e.isInstanceOf[GroupMaxSizeReachedException]) {
-                throw e
-              }
-              if (!kickedConsumerOut.compareAndSet(false, true)) {
-                fail(s"Received more than one ${classOf[GroupMaxSizeReachedException]}")
-              }
-              kickedOutConsumerIdx = Some(idx)
-          }
+      consumerPollers.indices.foreach(idx => {
+        consumerPollers(idx).thrownException match {
+          case Some(thrownException) =>
+            if (!thrownException.isInstanceOf[GroupMaxSizeReachedException]) {
+              throw thrownException
+            }
+            if (kickedConsumerOut) {
```

Comment:
Could probably get rid of this variable and just use `kickedOutConsumerIdx.foreach`.